src/repl.ml

```diff
@@ -0,0 +1,96 @@
+open Import
+
+let name = "REPL"
+
+let t = ref None
+
+let create_terminal sandbox =
+  match !t with
+  | Some term ->
+    Terminal_sandbox.show term;
+    Promise.return (Ok term)
+  | None -> (
+    let open Promise.Syntax in
+    let cmd = Sandbox.get_command sandbox "dune" [ "utop" ] in
+    let+ result = Cmd.check cmd in
+    match result with
+    | Error err -> Error err
+    | Ok (Shell _) ->
+      Error "Running a REPL from a shell command is not supported"
+    | Ok (Spawn cmd) ->
+      let term = Terminal_sandbox.create ~name ~command:cmd sandbox in
+      Terminal_sandbox.show term;
+      t := Some term;
+      Ok term )
+
+let get_code text_editor =
+  let selection = Vscode.TextEditor.selection text_editor in
+  let start_line = Vscode.Selection.start selection |> Vscode.Position.line in
+  let end_line = Vscode.Selection.start selection |> Vscode.Position.line in
+  let start_char =
+    Vscode.Selection.start selection |> Vscode.Position.character
+  in
+  let end_char =
+    Vscode.Selection.start selection |> Vscode.Position.character
+  in
+  if start_line = end_line && start_char = end_char then
+    let document = Vscode.TextEditor.document text_editor in
+    let line = Vscode.TextDocument.lineAt document ~line:start_line in
+    Vscode.TextLine.text line
+  else
+    let document = Vscode.TextEditor.document text_editor in
+    Vscode.TextDocument.getText document ~range:(selection :> Vscode.Range.t) ()
+
+let prepare_code code =
+  let end_chars = String.sub code ~pos:(String.length code - 2) ~len:2 in
```

Comment:
Base should have a function for suffix checking. Something like `String.is_suffix` I think.